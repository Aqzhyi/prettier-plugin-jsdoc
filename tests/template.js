/* eslint-disable no-undef */
const prettier = require('prettier')

function subject(code, options = {}) {
  return prettier.format(code, {
    parser: 'jsdoc-parser',
    plugins: ['.'],
    jsdocSpaces: 1,
    ...options,
  })
}

test('template for callback', () => {
  const result = subject(`
/**
 * @template T
 * @callback CallbackName
 * @param {GetStyles<T>} getStyles
 * @returns {UseStyle<T>}
 */
`)

  expect(result).toMatchSnapshot()
})
