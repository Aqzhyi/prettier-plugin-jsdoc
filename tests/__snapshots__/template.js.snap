// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template for callback 1`] = `
"/**
 * @template T
 * @callback CallbackName
 * @param {GetStyles<T>} getStyles
 * @returns {UseStyle<T>}
 */
"
`;
